stats/cloud/collector.go

```diff
@@ -287,14 +308,17 @@ func (c *Collector) Collect(sampleContainers []stats.SampleContainer) {
 	for _, sampleContainer := range sampleContainers {
 		switch sc := sampleContainer.(type) {
 		case *httpext.Trail:
-			sc = useCloudTags(sc)
+			sc = replaceTagsHTTP(sc)
+
 			// Check if aggregation is enabled,
 			if c.config.AggregationPeriod.Duration > 0 {
 				newHTTPTrails = append(newHTTPTrails, sc)
 			} else {
 				newSamples = append(newSamples, NewSampleFromTrail(sc))
 			}
 		case *netext.NetTrail:
+			sc = replaceTagsNet(sc)
```

Comment:
Wait, wat? :smile:  We don't need to replace any tags for `iter_li_all`/`netext.NetTrail`. The only tag we need to replace is `name`, and `name` only applies to HTTP URLs

Reply:
:D Right, I wondered about that, but considered we're also tagging network samples with the same tags as HTTP samples to associate them with the same HTTP context.

So is the test added in `TestCloudCollector` (https://github.com/loadimpact/k6/pull/1203/files#diff-11a7e92298be39c5269425095fb75dc9R283-R326) a fabricated scenario? Meaning `NetTrail` would never be tagged with `url` to begin with, right? Because that test was the reason for adding this replacement for `NetTrail` as well, but it seems unnecessary, or at least should use more realistic tags.

Let me know, but either way, I'll remove this replacement.

Reply:
Looking at the test now, it should probably be in its own separate test, not a part of the already huge `TestCloudCollector()`

> Meaning NetTrail would never be tagged with url to begin with, right? 

Yes, at least not by k6. If users are masochists, they might run k6 with global per-test-run manually set tags like `k6 run --tag name=blah --tag url=test script.js`, but we don't care about that. As a system tag, k6 only appends `url` to HTTP requests and websocket connections and `name` only for HTTP requests.

So yes, please remove the replacement and refactor the test.

Reply:
Done, see f032c5f6.

It might be a good idea to add tests for `Executor.Run` (to ensure we're _not_ sending the metric anymore) and for `Dialer.GetTrail`, but I didn't think it was strictly required, and being lazy again... But let me know if there should be more tests.